Dockerfile

```diff
@@ -1,31 +1,110 @@
       /etc/sudoers
+RUN USER=docker && \
+    GROUP=docker && \
+    curl -SsL https://github.com/boxboat/fixuid/releases/download/v0.3/fixuid-0.3-linux-amd64.tar.gz | tar -C /usr/local/bin -xzf - && \
+    chown root:root /usr/local/bin/fixuid && \
+    chmod 4755 /usr/local/bin/fixuid && \
+    mkdir -p /etc/fixuid && \
+    printf "user: $USER\ngroup: $GROUP\npaths:\n  - /service\n" > /etc/fixuid/config.yml
 
-USER ubuntu
+RUN echo "LANG=C.UTF-8" > /etc/default/locale
 
-ADD . /home/ubuntu/IncludeOS
-WORKDIR /home/ubuntu/IncludeOS
+# Docker TAG can be specified when building with --build-arg TAG=..., this is redeclared in the source-build stage
+# Git tags can be specified with --build-arg NEWTAG=..., the default is set as includeos-dev
+ARG BRANCH=dev
+ARG REPO=hioa-cs
+ARG NEWTAG=includeos-dev
+ENV BRANCH=$BRANCH
+ENV REPO=$REPO
+ENV NEWTAG=$NEWTAG
 
-RUN sudo apt-get update && \
-    sudo do_bridge="" ./etc/install_all_source.sh \
-&& sudo rm -rf /var/lib/apt/lists/*
 
-VOLUME /service
+LABEL dockerfile.version=1 \
+      includeos.version=$BRANCH
 WORKDIR /service
 
-CMD ./run.sh
+#########################
+FROM base as source-build
+
+RUN apt-get update && apt-get -y install \
+    git \
+    lsb-release \
+    net-tools \
+    wget \
+    && rm -rf /var/lib/apt/lists/*
+
+
+# Copy IncludeOS contents from host repo to container
+RUN echo "copying Contents of current IncludeOS branch"
+RUN cd ~ && pwd && \
+  mkdir -p IncludeOS
+COPY . /root/IncludeOS/
+
+# Adding Custom git tags
+RUN echo "Assigning Your custom git tag"
+RUN cd /root/IncludeOS && \
+    git tag -d $(git describe --tags) ; git tag $NEWTAG
+
+# Installation
+RUN cd /root/IncludeOS && \
+    ./install.sh -n
+
+RUN git -C /root/IncludeOS describe --dirty --tags > /ios_version.txt
+
+#############################
+FROM base as grubify
+
+RUN apt-get update && apt-get -y install \
+  dosfstools \
+  grub-pc
+
+COPY --from=source-build /usr/local/includeos/scripts/grubify.sh /home/ubuntu/IncludeOS_install/includeos/scripts/grubify.sh
+
+ENTRYPOINT ["fixuid", "/home/ubuntu/IncludeOS_install/includeos/scripts/grubify.sh"]
+
+###########################
+FROM base as build
+
+RUN apt-get update && apt-get -y install \
+    git \
+    clang-5.0 \
+    cmake \
+    nasm \
+    python-pip \
+    && rm -rf /var/lib/apt/lists/* \
+    && pip install pystache antlr4-python2-runtime && \
+    apt-get remove -y python-pip && \
+    apt autoremove -y
+
+COPY --from=source-build  /usr/local/includeos /usr/local/includeos/
+COPY --from=source-build  /usr/local/bin/boot /usr/local/bin/boot
+COPY --from=source-build  /root/IncludeOS/etc/install_dependencies_linux.sh /
+COPY --from=source-build  /root/IncludeOS/etc/use_clang_version.sh /
+COPY --from=source-build  /root/IncludeOS/lib/uplink/starbase /root/IncludeOS/lib/uplink/starbase/
+COPY --from=source-build  /ios_version.txt /
+COPY --from=source-build  /root/IncludeOS/etc/docker_entrypoint.sh /entrypoint.sh
```

Comment:
Are we missing the `etc/docker_entrypoint.sh` file?

Reply:
yes, will add it in the next PR